### PR TITLE
Add ClaimsEstablishment test data setup endpoint

### PR DESCRIPTION
**Pull request by shanear**
Add UploadableDocument concern to help with new documents

### CODE REVIEW

File: app/controllers/test/setup_controller.rb [hunk 1]
```diff
@@ -13,6 +13,19 @@ def certification
...
+  # Used for resetting data in UAT for claims establishment
+  def claims_establishment
+    # Only prepare test if there are less than 20 EstablishClaim tasks, as additional safeguard
+    fail "Too many ClaimsEstablishment tasks" if EstablishClaim.count > 20
+
+    EstablishClaim.delete_all
+    TestDataService.prepare_claims_establishment!(vacols_id: full_grant_id)
+    TestDataService.prepare_claims_establishment!(vacols_id: partial_grant_id, cancel_eps: true)
+
+    CreateEstablishClaimTasksJob.perform_now unless ApplicationController.dependencies_faked?
```
Comment (mdbenjam) [inline app/controllers/test/setup_controller.rb]: Do we want to do the stuff above if dependencies are faked? Or is it just this line?

File: app/services/appeal_repository.rb [hunk 2]
```diff
@@ -200,25 +198,28 @@ def self.uncertify(appeal)
   end
 
   def self.upload_form8(appeal, form8)
```
Comment (mdbenjam) [inline app/services/appeal_repository.rb]: Could we just make this, upload_and_delete_original to make it more generic?

File: app/controllers/test/setup_controller.rb (hunk 1 above)
Comment (mdbenjam) [inline app/controllers/test/setup_controller.rb]: Ah I see that prepare_claims_establishment! checks for the same thing. Maybe just put this entire method in that if?

File: spec/feature/test_setup_spec.rb [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    let(:appeal) { Appeal.create(vacols_id: "VACOLS123", vbms_id: "VBMS123") }
+
+    scenario "isn't allowed by a non-test user" do
+      User.authenticate!(roles: ["Establish Claim"])
+
+      visit "dispatch/establish-claim"
+      expect(page).not_to have_content("Reset Claims Establishment Tasks")
+    end
+
+    scenario "isn't allowed by a non-test user" do
```
Comment (mdbenjam) [inline spec/feature/test_setup_spec.rb]: Name should be is allowed.

File: app/views/tasks/worker_index.html.erb [hunk 4]
```diff
@@ -16,6 +16,10 @@
     <% end %>
   </div>
 
+  <% if test_user? %>
```
Comment (mdbenjam) [inline app/views/tasks/worker_index.html.erb]: Thoughts on putting all of these "dangerous operations" on one unified page?

Comment (shanear) [inline app/views/tasks/worker_index.html.erb]: We should follow up with @kierachell about this. Right now his integration tests look for links on this page.

File: app/controllers/test/setup_controller.rb (hunk 1 above)
Comment (shanear) [inline app/controllers/test/setup_controller.rb]: That would make some of the feature tests not work correctly

File: app/views/tasks/worker_index.html.erb (hunk 4 above)
Comment (mdbenjam) [inline app/views/tasks/worker_index.html.erb]: Cool, don't think this is necessary now, but might be good, going forward.

File: spec/feature/test_setup_spec.rb [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+    let(:appeal) { Appeal.create(vacols_id: "VACOLS123", vbms_id: "VBMS123") }
+
+    scenario "isn't allowed by a non-test user" do
+      User.authenticate!(roles: ["Establish Claim"])
+
+      visit "dispatch/establish-claim"
+      expect(page).not_to have_content("Reset Claims Establishment Tasks")
+    end
+
+    scenario "is allowed by a non-test user" do
```
Comment (mdbenjam) [inline spec/feature/test_setup_spec.rb]: nit: should be 'test user' not 'non-test user'

File: app/controllers/test/setup_controller.rb (hunk 1 above)
Comment (mdbenjam) [inline app/controllers/test/setup_controller.rb]: Got it, I see what you're saying.

File: app/views/tasks/worker_index.html.erb (hunk 4 above)
Comment (mdbenjam) [inline app/views/tasks/worker_index.html.erb]: Maybe consider making a tech improvement ticket if you think this is a good direction so we can keep it on our radar.

**Pull request merged**